src/net.cpp

```diff
@@ -2429,8 +2429,18 @@ bool CConnman::Start(CScheduler& scheduler, std::string& strNodeError, Options c
     }
 
 #ifndef WIN32
-    if (pipe2(wakeupPipe, O_NONBLOCK) != 0) {
+    if (pipe(wakeupPipe) != 0) {
```

Comment:
>O_NONBLOCK
              Set the O_NONBLOCK file status flag on the two new open file
              descriptions.  Using this flag saves extra calls to fcntl(2)
              to achieve the same result.

Maybe it would be better to use `pipe` for macos only and keep `pipe2` for other cases besides win32 i.e. smth like
```
#ifdef MAC_OSX
    if (pipe(wakeupPipe) != 0) {
        wakeupPipe[0] = wakeupPipe[1] = -1;
        LogPrint("net", "pipe() for wakeupPipe failed\n");
    } else {
        int fFlags = fcntl(wakeupPipe[0], F_GETFL, 0);
        if (fcntl(wakeupPipe[0], F_SETFL, fFlags | O_NONBLOCK) == -1) {
            LogPrint("net", "fcntl for O_NONBLOCK on wakeupPipe failed\n");
        }
        fFlags = fcntl(wakeupPipe[1], F_GETFL, 0);
        if (fcntl(wakeupPipe[1], F_SETFL, fFlags | O_NONBLOCK) == -1) {
            LogPrint("net", "fcntl for O_NONBLOCK on wakeupPipe failed\n");
        }
    }
#elif !defined(WIN32)
    if (pipe2(wakeupPipe) != 0) {
        wakeupPipe[0] = wakeupPipe[1] = -1;
        LogPrint("net", "pipe2() for wakeupPipe failed\n");
    }
#endif
```

Reply:
Hmm, why? As I understand it from the man pages, `pipe2` with `O_NONBLOCK` is really the same as what I did to make it work on Mac OS.

Reply:
>Using this flag saves extra calls to fcntl(2) to achieve the same result.

No sure how much we can win here though.

Reply:
This is a one-time thing on startup, so really not worth the effort of maintaining two versions of the code for two platforms. In such cases, it's IMHO better to stick the common solution, even if it's slightly suboptimal.

Reply:
Ah, true. I completely missed the fact that it's only a startup thing.